src/__tests__/cli.ts

```diff
@@ -1,32 +1,61 @@
 import fs from 'fs';
 import path from 'path';
 import util from 'util';
-import childProcess from 'child_process';
+import { CliArguments } from '..';
 
 const mkdir = util.promisify(fs.mkdir);
 const rmdir = util.promisify(fs.rmdir);
 const writeFile = util.promisify(fs.writeFile);
 const readFile = util.promisify(fs.readFile);
 
-function exec(
-  command: string,
-  options: { cwd: string },
-): Promise<{ exitCode: number | null; stdout: string; stderr?: string }> {
-  return new Promise((resolve) => {
-    try {
-      const buffer = childProcess.execSync(command, options);
-      resolve({
-        exitCode: 0,
-        stdout: buffer.toString(),
-      });
-    } catch (ex) {
-      resolve({
-        exitCode: ex.status,
-        stdout: ex.stdout.toString(),
-        stderr: ex.stderr.toString(),
-      });
-    }
-  });
+async function exec(
+  testProjectDir: string,
+  { init = false, flow = false, update = false }: Partial<CliArguments> = {},
+): Promise<{ exitCode: number | null; stdout: string; stderr: string }> {
+  const originalExit = process.exit;
+  const originalCwd = process.cwd();
+  const originalConsole = {
+    log: console.log,
+    warn: console.warn,
+    error: console.error,
+  };
+
+  try {
+    let exitCode: number | null = null;
+    let stdout = '';
+    let stderr = '';
+
+    const appendStdout = (...args: any[]): void => {
+      stdout += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    const appendStderr = (...args: any[]): void => {
+      stderr += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    console.log = appendStdout;
+    console.warn = appendStdout;
+    console.error = appendStderr;
```

Comment:
capture console output for assertions.  Not sure if it's worth going overboard here and including things like `console.assert` and `console.trace`?